### PR TITLE
Add tags and read posts to frontpage user recommendations

### PR DESCRIPTION
**Pull request by lagerros**
This PR includes: 
* Show a user's top commented tag in their frontpage reciprocity widget
* Show posts you've read by a user in the reciprocity widget
* If we can't fetch enough recommended users meeting our standard criteria, relax the criteria to fetch more users 
* Refactor the widget to make it easier to recommend and show different kinds of topic ideas
* Increase the A/B test skew to show the widget to more users, now that it has been improved

image

image

image

### CODE REVIEW

File: packages/lesswrong/components/dialogues/DialogueRecommendationRow.tsx [hunk 1]
```diff
@@ -182,40 +189,90 @@ const DialogueRecommendationRow = ({ rowProps, classes, showSuggestedTopics }: D
...
+        _id
+        title
+      }
+    }
+  `, {
+    variables: { userId: currentUser?._id, targetUserId: targetUser._id, limit : 3 },
+  });
+
+  const topTags:[TagWithCommentCount] = tagData?.UserTopTags;
+  const readPosts:DbPost[] = postsData?.UsersReadPostsOfTargetUser
```
Comment (b0b3rt) [inline packages/lesswrong/components/dialogues/DialogueRecommendationRow.tsx:218]: these are not `DbPost`; that's a purely server-side and objects on the client will never actually fit that type

File: packages/lesswrong/components/dialogues/DialogueRecommendationRow.tsx [hunk 2]
```diff
@@ -182,40 +189,90 @@ const DialogueRecommendationRow = ({ rowProps, classes, showSuggestedTopics }: D
...
   const preTopicRecommendations: TopicRecommendationWithContents[] | undefined = topicData?.GetTwoUserTopicRecommendations; 
   const topicRecommendations = preTopicRecommendations?.filter(topic => ['agree', 'disagree'].includes(topic.theirVote) ); // todo: might want better type checking here in future for values of theirVote
-  const numRecommendations = topicRecommendations?.length ?? 0;
-  const numShown = isExpanded ? numRecommendations : 1
+ 
+  if (!currentUser || !topTags || !topicRecommendations || !readPosts) return <></>;
+  const tagsSentence = topTags.slice(0, 4).map(tag => tag.tag.name).join(', ');
+  const numRecommendations = topicRecommendations?.length + readPosts?.length + 1 ?? 0;
```
Comment (b0b3rt) [inline packages/lesswrong/components/dialogues/DialogueRecommendationRow.tsx:224]: The null coalesce isn't ever going to do anything if you add 1 first, but if you try adding 1 to `undefined` I think you get `NaN`.  (So you need to do the coalesce before adding.)

File: packages/lesswrong/components/dialogues/DialogueRecommendationRow.tsx [hunk 3]
```diff
@@ -182,40 +189,90 @@ const DialogueRecommendationRow = ({ rowProps, classes, showSuggestedTopics }: D
...
+  const getTopicSuggestion = (reactIconName:string, prefix:string, Content:JSX.Element) => {
+    return (
+      <div className={classes.suggestionRow}>
+        <p className={classNames({
+          [classes.debateTopicExpanded]: isExpanded,
+          [classes.debateTopicCollapsed]: !isExpanded
+        })}>
+            <ReactionIcon key={"1"} size={13} react={reactIconName} />
+            {" "}
+            <span key={"2"} className={classNames(classes.agreeText, { [classes.agreeTextCollapsedMobile]: !isExpanded})}>
```
Comment (b0b3rt) [inline packages/lesswrong/components/dialogues/DialogueRecommendationRow.tsx:237]: nit: attributes which have fixed string values (i.e. `"1"`, `"2"`) can just be written like so: `key="1"`

Comment (b0b3rt) [reply packages/lesswrong/components/dialogues/DialogueRecommendationRow.tsx:292]: @lagerros I roll to disbelieve that it saves you time to fix things like this in follow-up PRs.

Comment (lagerros) [reply packages/lesswrong/components/dialogues/DialogueRecommendationRow.tsx:292]: for sure, just fixed it in this one

**Pull request merged**